lib/mail/field.rb

```diff
@@ -232,12 +232,13 @@ def split(raw_field)
     #  that is immediately followed by WSP.  Each header field should be
     #  treated in its unfolded form for further syntactic and semantic
     #  evaluation.
-    def unfold(string)
-      string.gsub(/[\r\n \t]+/m, ' ')
+    def unfold(name, value)
+      regex = name == 'Subject' ? /\s*\r\n\s+/m : /[\r\n \t]+/m
```

Comment:
I'd expect the "unfolding" there to only remove the spaces after the \r\n
and before Lindsaar. Is there an rfc that specifies this case?

On April 5, 2016 at 7:04:28 PM, andrew selder (notifications@github.com)
wrote:

> In lib/mail/field.rb
> <https://github.com/mikel/mail/pull/981#discussion_r58634519>:
>
> > @@ -232,12 +232,13 @@ def split(raw_field)
> >      #  that is immediately followed by WSP.  Each header field should be
> >      #  treated in its unfolded form for further syntactic and semantic
> >      #  evaluation.
> > -    def unfold(string)
> > -      string.gsub(/[\r\n \t]+/m, ' ')
> > +    def unfold(name, value)
> > +      regex = name == 'Subject' ? /\s*\r\n\s+/m : /[\r\n \t]+/m
>
> @kjg <https://github.com/kjg>. That's quite possible, that the specs are
> wrong. But I wanted to cause as little disruption as possible.
>
> An example it the 'To' header, in the spec right above the 2 I added:
>
>     it "should remove multiple spaces during unfolding a header" do
>       header = Mail::Header.new("To: Mikel,\r\n   Lindsaar,     Bob")
>       expect(header['To'].value).to eq 'Mikel, Lindsaar, Bob'
>     end
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mikel/mail/pull/981/files/c93e25218db01b6177a34323b58bde30684ce57b#r58634519>
>